Pcap++/header/PcapRemoteDeviceList.h

```diff
@@ -26,37 +27,36 @@ namespace pcpp
 	 * by iterating the PcapRemoteDevice instances (through the PcapRemoteDeviceList#RemoteDeviceListIterator iterator)<BR>
 	 * Since Remote Capture is supported in WinPcap and Npcap only, this class is available in Windows only
 	 */
-	class PcapRemoteDeviceList
+	class PcapRemoteDeviceList : public internal::DeviceListBase<PcapRemoteDevice>
 	{
 	private:
-		std::vector<PcapRemoteDevice*> m_RemoteDeviceList;
+		using Base = internal::DeviceListBase<PcapRemoteDevice>;
+
 		IPAddress m_RemoteMachineIpAddress;
 		uint16_t m_RemoteMachinePort;
 		std::shared_ptr<PcapRemoteAuthentication> m_RemoteAuthentication;
 
 		// private c'tor. User should create the list via static methods PcapRemoteDeviceList::createRemoteDeviceList()
 		PcapRemoteDeviceList(const IPAddress& ipAddress, uint16_t port,
 		                     std::shared_ptr<PcapRemoteAuthentication> remoteAuth,
-		                     std::vector<PcapRemoteDevice*> deviceList);
+		                     PointerVector<PcapRemoteDevice> deviceList);
 
 	public:
 		/**
 		 * Iterator object that can be used for iterating all PcapRemoteDevice in list
 		 */
-		using RemoteDeviceListIterator = typename std::vector<PcapRemoteDevice*>::iterator;
+		using RemoteDeviceListIterator = iterator;
 
 		/**
 		 * Const iterator object that can be used for iterating all PcapRemoteDevice in a constant list
 		 */
-		using ConstRemoteDeviceListIterator = typename std::vector<PcapRemoteDevice*>::const_iterator;
+		using ConstRemoteDeviceListIterator = const_iterator;
```

Comment:
Maybe we can mark them as deprecated? User can use `iterator` and `const_iterator`

Reply:
Deprecated them in cee1e43

Reply:
Maybe you should put `PCPP_DEPRECATED` above the `using` line instead of inline, and then MinGW will compile? 🤔

Reply:
Thinking about it again - not sure putting the macro above the `using` statement will do what we want, but I'm not 100% sure... Worst case we can either drop `RemoteDeviceListIterator` and `ConstRemoteDeviceListIterator` completely, even though it's a breaking change (I don't believe many people use it anyway), or just keep it without deprecation

Reply:
Ok, had to break out my voodoo doll but managed to get it working in bb695b8.